components/camel-seda/src/main/java/org/apache/camel/component/seda/SedaConsumer.java

```diff
@@ -215,15 +210,38 @@ protected void doRun() {
         }
     }
 
+    /**
+     * Strategy to invoke when the exchange is done being processed.
+     * <p/>
+     * This method is meant to be overridden by subclasses to be able to mimic the behavior of the legacy component
+     * camel-vm, that is why the parameter {@code prepared} is not used by default.
+     *
+     * @param original the exchange before being processed
+     * @param prepared the exchange processed
+     */
+    protected void onProcessingDone(Exchange original, Exchange prepared) {
```

Comment:
You do not need to have 2 parameters here, it can just be named exchange, and it should be the `prepared` exchange that is being called from the callback.

Reply:
Interesting feedback, I was not sure TBH, I just wanted to allow to call `copyResults` like before the change https://github.com/apache/camel/commit/ed5cc8a6c9f54a4fdf25a2eed03fa528c8b0f9a1#diff-495571c4ab51f8270c10e0c9d149c351014465883804a481440403ebe4e4fc4dL190, it is not needed anymore?

Reply:
ah yeah that is actually needed since this logic is now in the callback (async compatible) instead of afterwards sendToConsumers.